zutils.cc

```diff
@@ -23,7 +23,65 @@ void ZBackup::backupFromStdin( string const & outputFileName )
 {
   if ( isatty( fileno( stdin ) ) )
     throw exWontReadFromTerminal();
+  backupFromFileHandle( "stdin", stdin, outputFileName );
+}
+
+/// Backs up the data from a file
+void ZBackup::backupFromFile( string const & inputFileName, string const & outputFileName )
+{
+  FILE* f = fopen( inputFileName.c_str(), "r" );
+  if ( NULL == f )
```

Comment:
Special: You mean you didn't want to backup /dev/urandom?  :-)  Will address this.

Symlink: Have changed the file open code in zutil.cc to not use fopen() directly, but to use the File class.  With this in place, valid symlinks are followed (I consider this good behavior), and dangling symlinks emit the error shown below, which I also consider to be useful enough.  If I have misunderstood some other sublety regarding your comment, please let me know.

<pre>
./zbackup backup --non-encrypted /tmp/foo /dev/shm/ooo/backups/111/
Loading index...
Loading index file f9ff8da1f6120784ba953f95b1010313174d1f2d77dfb0d3...
Index loaded.
Using up to 4 thread(s) for compression
Can't open /tmp/foo/asdfaksjdf: No such file or directory
</pre>